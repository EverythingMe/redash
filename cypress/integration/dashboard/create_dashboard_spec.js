function createNewDashboard(dashboardName) {
  cy.getByTestId('CreateButton').click();
  cy.get('li[role="menuitem]"').contains('Dashboard').click();
  cy.getByTestId('EditDashboardDialog').within(() => {
    cy.getByTestId('DashboardSaveButton').should('be.disabled');
    cy.get('input').type(dashboardName);
    cy.getByTestId('DashboardSaveButton').click();
  });
}

function archiveCurrentDashboard() {
  cy.getByTestId('DashboardMoreMenu').click().within(() => {
    cy.get('li').contains('Archive').click();
  });

  cy.get('.btn-warning').contains('Archive').click();
  cy.get('.label-tag-archived').should('exist');
}

describe('Dashboard', () => {
  beforeEach(() => {
    cy.login();
    cy.visit('/dashboards');
  });

  it('creates a new dashboard and archives it', () => {
    cy.getByTestId('DashboardLayoutContent').as('content');
    cy.get('@content').toMatchSnapshot(); // TODO: make this overwrite https://git.io/fhjRW

    createNewDashboard('Foo Bar');

    let urlPath;
    cy.location().should(({ pathname }) => {
      urlPath = pathname;
      expect(pathname).to.include('dashboard/foo-bar');
    });

    cy.visit('/dashboards');
    cy.get('@content').within(() => {
      cy.get('.table-main-title')
      .should('exist')
      .and('have.attr', 'href', urlPath.substring(1))
      .click();
    });

    archiveCurrentDashboard();

    cy.visit('/dashboards');
    cy.get('@content').toMatchSnapshot();
  });
});
