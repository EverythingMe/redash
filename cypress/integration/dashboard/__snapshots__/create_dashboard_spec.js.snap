exports[`Dashboard > creates a new dashboard and archives it #0`] = `
<div data-test="DashboardLayoutContent">
  <div class="empty-state bg-white tiled">
    <div class="empty-state__summary">
      <h2><i class="zmdi zmdi-view-quilt"></i></h2>
      <p>See the big picture</p>
      <img
        src="/static/images/illustrations/dashboard.svg"
        alt="dashboard Illustration"
        width="75%"
      />
    </div>
    <div class="empty-state__steps">
      <h4>Let's get started</h4>
      <ol>
        <li class="done">
          <a href="data_sources/new">Connect</a> a Data Source
        </li>
        <li class=""><a href="queries/new">Create</a> your first Query</li>
        <li class="">
          <a href="javascript:void(0)">Create</a> your first Dashboard
        </li>
      </ol>
      <p>
        Need more support?
        <a
          href="https://help.redash.io/category/22-dashboards"
          target="_blank"
          rel="noopener noreferrer"
          >See our Help<i
            class="fa fa-external-link m-l-5"
            aria-hidden="true"
          ></i
        ></a>
      </p>
    </div>
  </div>
</div>
`;

exports[`Dashboard > creates a new dashboard and archives it #1`] = `
<div data-test="DashboardLayoutContent">
  <div class="empty-state bg-white tiled">
    <div class="empty-state__summary">
      <h2><i class="zmdi zmdi-view-quilt"></i></h2>
      <p>See the big picture</p>
      <img
        src="/static/images/illustrations/dashboard.svg"
        alt="dashboard Illustration"
        width="75%"
      />
    </div>
    <div class="empty-state__steps">
      <h4>Let's get started</h4>
      <ol>
        <li class="done">
          <a href="data_sources/new">Connect</a> a Data Source
        </li>
        <li class=""><a href="queries/new">Create</a> your first Query</li>
        <li class="">
          <a href="javascript:void(0)">Create</a> your first Dashboard
        </li>
      </ol>
      <p>
        Need more support?
        <a
          href="https://help.redash.io/category/22-dashboards"
          target="_blank"
          rel="noopener noreferrer"
          >See our Help<i
            class="fa fa-external-link m-l-5"
            aria-hidden="true"
          ></i
        ></a>
      </p>
    </div>
  </div>
</div>
`;
